docker/docker_image.go

```diff
@@ -39,25 +41,70 @@ func (i *Image) SourceRefFullName() string {
 	return i.src.ref.ref.Name()
 }
 
-// GetRepositoryTags list all tags available in the repository. Note that this has no connection with the tag(s) used for this specific image, if any.
+// GetRepositoryTags list all tags available in the repository. Note that this
+// has minimal connection with the tag(s) used for this specific
+// ImageReference; the tag will be used to determine policy-based access (see
+// PolicyConfigurationIdentity), but will otherwise be ignored. (This is a
+// backward-compatible shim method which calls the module-level GetRepositoryTags)
 func (i *Image) GetRepositoryTags(ctx context.Context) ([]string, error) {
-	path := fmt.Sprintf(tagsPath, reference.Path(i.src.ref.ref))
-	// FIXME: Pass the context.Context
-	res, err := i.src.c.makeRequest(ctx, "GET", path, nil, nil)
-	if err != nil {
-		return nil, err
-	}
-	defer res.Body.Close()
-	if res.StatusCode != http.StatusOK {
-		// print url also
-		return nil, errors.Errorf("Invalid status code returned when fetching tags list %d", res.StatusCode)
+	return GetRepositoryTags(ctx, i.src.c.sys, i.src.ref)
+}
+
+// GetRepositoryTags list all tags available in the repository. Note that this
+// has minimal connection with the tag(s) used for this specific
+// ImageReference; the tag will be used to determine policy-based access (see
+// PolicyConfigurationIdentity), but will otherwise be ignored.
```

Comment:
I’m not sure what “determine policy-based access” refers to; the only use of `.Tag()` invoked through this which I can see is in `lookaside.go`, and yes, that involves `PolicyConfigurationIdentity`, but ultimately the `signatureBase` value is not used for anything in this function.  Am I overlooking something?

(Applies to the shim as well.)

Reply:
I admit I don't understand the policy stuff much, that comment was an attempt to explain why a tagged reference has to be passed in even though the tag is ignored, perhaps it could be better stated somehow?